core/src/main/java/com/linecorp/armeria/server/HttpResponseSubscriber.java

```diff
@@ -200,7 +200,11 @@ public void onNext(HttpObject o) {
                                 " (service: " + service() + ')'));
                         return;
                     }
-                    setDone(true);
+
+                    // Mark as DONE and send a request for receiving onComplete() or canceling the upstream
+                    setDone(false);
+                    subscription.request(1);
```

Comment:
`subscription.request(1)` is called in `writeHeadersFutureListener` so I think we don't need to call this here.

Reply:
Good point! I missed that.